release-2.0: rpc: adopt logging in circuitbreaker

ajwerner opened the pull request:
Backport 1/1 commits from #33676.

/cc @cockroachdb/release

---

Adopts changes to the circuitbreaker package to enable logging. 

Release note: None

Commit:
rpc: adopt logger in circuit breaker

Adopts changes to the circuitbreaker package to enable logging.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/35361)
<!-- Reviewable:end -->

andreimatei reviewed (commented):
Is our copy of the circuitbreaker package pretty much identical to the dependency you're removing here?

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/35361#-:-L_9TZMaAyPKK7Lc5T1l:bkijugr)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @andreimatei)



<!-- Sent from Reviewable.io -->

ajwerner commented:
> Is our copy of the circuitbreaker package pretty much identical to the dependency you're removing here?
> 
> > _[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/35361#-:-L_9TZMaAyPKK7Lc5T1l:bkijugr)_ status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @andreimatei)

It's a fork of that library. We've fixed some bugs and added more logging. I think we're about 4 commits ahead of the old dependency which seems to be abandoned.

andreimatei commented:
LGTM
